### ISSUE TITLE
Compatibility issues with fig autocomplete

### ISSUE THREAD

**Issue by jwyce** [Schniz/fnm#927]
Hi @Schniz,

This is by far my favorite node management tool and I've been using it for almost a year!

In the last couple of months or so I've noticed a strange issue where the `fnm use <version>` command no longer worked, but `fnm install` and `fnm default` all worked as expected.

![image](https://user-images.githubusercontent.com/16946573/218265915-b952e36d-abbf-4ede-b62c-51f05f49e4a2.png)

however, when I manually eval fnm before using `fnm use` it works just fine.

my trimmed down .zshrc looks like this 
```shell
# Fig pre block. Keep at the top of this file.
[[ -f "$HOME/.fig/shell/zshrc.pre.zsh" ]] && builtin source "$HOME/.fig/shell/zshrc.pre.zsh"
# Load Antigen
source "$HOME/antigen.zsh"

# Load Antigen configurations
antigen init ~/.antigenrc

export DOT=$HOME/.dotfiles
export ZSHRC=$DOT/.zshrc
export BIN=$DOT/bin/scripts
export PATH="$BIN:$PATH"

eval "$(fnm env --use-on-cd)"
eval "$(starship init zsh)"

# Enable vi mode
bindkey -v

# rebind atuin
ATUIN_NOBIND=1 eval "$(atuin init zsh)"
bindkey '^r' _atuin_search_widget

export PATH="$PATH:/Users/wabamn/.bin"
# Fig post block. Keep at the bottom of this file.
[[ -f "$HOME/.fig/shell/zshrc.post.zsh" ]] && builtin source "$HOME/.fig/shell/zshrc.post.zsh"
```

and I have the following .zprofile

```shell
# Fig pre block. Keep at the top of this file.
[[ -f "$HOME/.fig/shell/zprofile.pre.zsh" ]] && builtin source "$HOME/.fig/shell/zprofile.pre.zsh"
eval "$(/opt/homebrew/bin/brew shellenv)"

# Fig post block. Keep at the bottom of this file.
[[ -f "$HOME/.fig/shell/zprofile.post.zsh" ]] && builtin source "$HOME/.fig/shell/zprofile.post.zsh"
```

Also what I've found is when I remove or comment out the fig (https://fig.io/) pre and post blocks fnm works just fine. ~I'm planning on making a cross-issue over in the fig repo as well to see if it's a problem on their side~ (see cross-issue here: https://github.com/withfig/fig/issues/2331) , but was wondering if you had any ideas on what could be going wrong here?

**Comment by Schniz** [comment]
Woah, I actually have no idea! I wonder what magical stuff they're doing. I'm not using fig though. Would be awesome to learn why it fails, so please share if you debug it <picture data-single-emoji=":pray_parrot:" title=":pray_parrot:"><img class="emoji" width="20" height="auto" src="https://single-emoji.vercel.app/api/emoji/eyJhbGciOiJkaXIiLCJlbmMiOiJBMjU2R0NNIn0..A_TBujSitbbN8fxV.uRH8BvYjvZPozrXfepmfyH225A5cdgBO87btHRVRlVWpCPfR4TMckxICbHdikQB82t34wbvz0BYWB7dLCIxI3acjtHQ5-mwCzdkFFrw61-VUag857DcWc5Wtrxo.PMoOpzOQv5hx8DWaCpfwIQ" alt=":pray_parrot:" align="absmiddle"></picture> (what fnm does is pretty basic, so I don't understand how fig can make it fail)

would be helpful if you also did `env | grep FNM_` so we see what is _currently_ configured

**Comment by jwyce** [comment]
I'll definitely share when/if I find a solution! Here is that output too

![image](https://user-images.githubusercontent.com/16946573/218329756-9868252b-2048-43ea-8ccb-c300ebb20d92.png)

**Comment by jwyce** [comment]
So I think I found my issue. fig does some weird tracking of your shell environment and caches it in `/Users/<name>/Libray/Application Support/fig/shells` and the state.json file one directory up. 

I found this out when setting up a new computer with a different username and when I ran `fnm env` the user for my old computer was in the `FNM_DIR`. Manually doing a find a replace for the new user name fixed it for my new computer, but I'm thinking maybe doing a reinstall of fnm got out of sync with fig's caching. 

Anyways I'm closing this for now, but hopefully it helps others if they run into a similar issue!
